packages/browser/src/index.test.ts

```diff
@@ -1,17 +1,51 @@
-import { createRequester } from '@logto/js';
+import { generateCodeChallenge, parseUriParameters, QueryKey } from '@logto/js';
 
-import LogtoClient from './index';
+import LogtoClient, { getLogtoKey, LogtoSignInSessionItem } from './index';
+
+const clientId = 'client_id_value';
+const endpoint = 'https://logto.dev';
+const authorizationEndpoint = `${endpoint}/oidc/auth`;
+const requester = jest.fn();
+
+jest.mock('@logto/js', () => ({
+  ...jest.requireActual('@logto/js'),
+  fetchOidcConfig: async () => ({
+    authorizationEndpoint,
+  }),
+}));
 
 describe('LogtoClient', () => {
   test('constructor', () => {
-    expect(
-      () =>
-        new LogtoClient({
-          endpoint: 'https://logto.dev',
-          clientId: 'client_id_value',
-          usingPersistStorage: false,
-          requester: createRequester(),
-        })
-    ).not.toThrow();
+    expect(() => new LogtoClient({ endpoint, clientId, requester })).not.toThrow();
+  });
+
+  describe('signInWithBrowser', () => {
+    test('sessionStorage and signInUri should contain correct properties', async () => {
+      const logtoClient = new LogtoClient({ endpoint, clientId, requester });
+      const originalRedirectUri = 'http://localhost:3000/callback';
+      await logtoClient.signInWithBrowser(originalRedirectUri);
+
+      const item = sessionStorage.getItem(getLogtoKey(clientId));
+      expect(item).not.toBeNull();
+
+      if (item) {
+        const logtoSignInSessionItem = JSON.parse(item) as LogtoSignInSessionItem;
+        const { state, codeVerifier, redirectUri } = logtoSignInSessionItem;
+        const codeChallenge = await generateCodeChallenge(codeVerifier);
+
+        expect(redirectUri).toEqual(originalRedirectUri);
+        expect(state.length).toBeGreaterThan(1);
+        expect(codeVerifier.length).toBeGreaterThan(1);
```

Comment:
nit: something like `.toBeGreaterThan(1).toLessOrEqualTo(128)`?

Reply:
seems to be better test these cases (length<=128) for `generateState` and `generateCodeVerifier` in `js/js`?
(I just intend to test they are not blank here.)

Reply:
then we should not assert anything here? since they all should be covered in js unit tests

Reply:
Finally

```ts
        expect(codeVerifier.length).toBeGreaterThan(1);
        expect(codeVerifier.length).toBeLessThanOrEqual(128);
```

since `expect(…).toBeXXX(…)` calling cannot be chained.

Reply:
The original code no longer exists.
Resolved this comment.